elastic/kibana
# [Fleet] use package storage when getting a package

This is the implementation of integrations using the [package storage](https://github.com/elastic/kibana/pull/83391) to get assets or install from.

- adds function `getEsPackage` to get the package in storage.  this function has to create the 'package info' as well.
- modifies calls to get the package to look in es storage if it doesn't exist in cache

Manual test uploaded package
- upload a package
- make a change to the server so the package doesnt exist in the cache
- GET /api/fleet/epm/packages/{pkg} should return package info

Manual test package installed from registry
- install package from registry
- make a change to the server so the package doesnt exist in the cache
- GET /api/fleet/epm/packages/{pkg} should return package info

Manual test  package installed from registry before the storage feature was available
- install package from registry
- delete all the items related to this package in elasticsearch storage so it doesn't exist there
- make a change to the server so the package doesnt exist in the cache
- GET /api/fleet/epm/packages/{pkg} should return package info by reaching out to the registry
